src/main/scala/com/databricks/spark/sql/perf/mllib/feature/VectorAssembler.scala

```diff
@@ -0,0 +1,49 @@
+package com.databricks.spark.sql.perf.mllib.feature
+
+import org.apache.spark.ml
+import org.apache.spark.ml.linalg.Vector
+import org.apache.spark.ml.PipelineStage
+import org.apache.spark.sql._
+import com.databricks.spark.sql.perf.mllib.OptionImplicits._
+import com.databricks.spark.sql.perf.mllib.data.DataGenerator
+import com.databricks.spark.sql.perf.mllib.{BenchmarkAlgorithm, MLBenchContext, TestFromTraining}
+
+/** Object for testing VectorAssembler performance */
+object VectorAssembler extends BenchmarkAlgorithm with TestFromTraining {
+
+  val numInputCols = 3
```

Comment:
Here use both `numFeatures` and `numInputCols`, representing input contains `numInputCols` column, and each column is a vector of size `numFeatures`.
I add `numInputCols` into `MLParam`.